src/main/scala/scala/collection/decorators/IteratorDecorator.scala

```diff
@@ -5,6 +5,20 @@ import scala.annotation.tailrec
 
 class IteratorDecorator[A](val `this`: Iterator[A]) extends AnyVal {
 
+  /**
+    * Inserts a separator value between each element.
+    *
+    * {{{
+    *   Iterator(1, 2, 3).intersperse(0) === Iterator(1, 0, 2, 0, 3)
```

Comment:
I think that if we give code examples, then this should be code examples that can be copy-pasted. Here the method `===` does not exist, and even `==` wouldn’t work.

Maybe we can just add a note at the beginning explaining that this is “pseudo-code” and that `===` means “is equivalent as if we had written”?

Reply:
We could write `assert( ... == ... )`, that would be executable.

Reply:
Or just `.. == ..`. That would executable also :)

Reply:
Yes, but `==` is reference equality on `Iterator`, not value equality, so, executing that code would return `false`.

Reply:
Or: `Iterator(1, 2, 3).intersperse(0) // equals Iterator(1, 0, 2, 0, 3)`

Reply:
I'll take your suggestion.

Reply:
Fixed.